homeassistant/components/upnp/binary_sensor.py

```diff
@@ -0,0 +1,108 @@
+"""Support for UPnP/IGD Binary Sensors."""
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import Any, Mapping
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_CONNECTIVITY,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import (
+    CONFIG_ENTRY_SCAN_INTERVAL,
+    CONFIG_ENTRY_UDN,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    DOMAIN_DEVICES,
+    LOGGER,
+    UPTIME,
+    WANIP,
+    WANSTATUS,
+)
+from .device import Device
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the UPnP/IGD sensors."""
+    udn = config_entry.data[CONFIG_ENTRY_UDN]
+    device: Device = hass.data[DOMAIN][DOMAIN_DEVICES][udn]
+
+    update_interval_sec = config_entry.options.get(
+        CONFIG_ENTRY_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+    )
+    update_interval = timedelta(seconds=update_interval_sec)
+    LOGGER.debug("update_interval: %s", update_interval)
+    LOGGER.debug("Adding sensors")
+    coordinator = DataUpdateCoordinator[Mapping[str, Any]](
+        hass,
+        LOGGER,
+        name=device.name,
+        update_method=device.async_get_status,
+        update_interval=update_interval,
+    )
+    device.coordinator = coordinator
+
+    await coordinator.async_refresh()
+
+    sensors = [
+        UpnpStatusBinarySensor(coordinator, device),
+    ]
+    async_add_entities(sensors)
+
+
+class UpnpStatusBinarySensor(CoordinatorEntity, BinarySensorEntity):
+    """Base class for UPnP/IGD binary sensors."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator[Mapping[str, Any]],
+        device: Device,
+    ) -> None:
+        """Initialize the base sensor."""
+        super().__init__(coordinator)
+        self._device = device
+        self._attr_name = f"{device.name} wan status"
+        self._attr_unique_id = f"{device.udn}_wanstatus"
+        self._attr_device_info = {
+            "connections": {(dr.CONNECTION_UPNP, device.udn)},
+            "name": device.name,
+            "manufacturer": device.manufacturer,
+            "model": device.model_name,
+        }
+        self._attr_device_class = DEVICE_CLASS_CONNECTIVITY
```

Comment:
This is static and should be defined as a class attribute directly on the `UpnpStatusBinarySensor` class

```
class UpnpStatusBinarySensor(CoordinatorEntity, BinarySensorEntity):
    """Base class for UPnP/IGD binary sensors."""

    _attr_device_class = DEVICE_CLASS_CONNECTIVITY
```

Reply:
It's already specified in `manifest.json`, is it really needed to add it here too?

Reply:
The  [`iot_class`](https://developers.home-assistant.io/docs/creating_integration_manifest/#iot-class) is defined in `manifest.json`, which has nothing to do with the entity presentation (which the `device_class` does)